### PR TITLE
Previews for .epub, .cbz, and .cbr files.

### PR DESCRIPTION
**Pull request by NoMoreNicksLeft**
I expect that before these could be merged that I'd have to do some minor code cleanup. If anyone could give me some pointers towards that.

There is also the issue of the mime types for digital comic books. NC currently marks them all as cbr regardless of the actual container (and there are apparently more beyond just zip and rar). Logically they are simply different types of files that use a similar strategy. But IETF is never going to give them an official mimetype either. The naming convention for modern mime-types strongly suggests that these end in +zip and +rar, as with epub and svg, thus leaving the only real issue as the first part of the mime. I've got that as "comicbook", but it could be anything reasonable. 

x- in front of the name is deprecated in favor of "vnd." now. That just makes it uncomfortably complicated (vnd.comicbook+zip) in my opinion, and doesn't really do anything. I don't believe that this can cause problems with other software, there's no real way for it to contaminate mimetypes in some other system. 

That said, it's easy to change if anyone insists that it be different.

### CODE REVIEW

**Comment by rullzer** [comment]
@NoMoreNicksLeft cool stuff.

I'm a bit torn what to do here. On the one hand I want this kind of stuff in apps (I even want the main providers in an app shipped by default).

This would however require also work in letting apps register mimetypes etc.
I'll discuss with some other people what the best approach is here.

**Comment by MorrisJobke** [comment]
> This would however require also work in letting apps register mimetypes etc.
> I'll discuss with some other people what the best approach is here.

It makes sense to keep the mime type update in the server and extract the actual preview provider in a separate app that provides them.

File: resources/config/mimetypemapping.dist.json [hunk 1]
```diff
@@ -21,12 +21,12 @@
...
-	"cbr": ["application/x-cbr"],
-	"cbt": ["application/x-cbr"],
-	"cbtc": ["application/x-cbr"],
-	"cbz": ["application/x-cbr"],
+	"cb7": ["application/comicbook+7z"],
+	"cba": ["application/comicbook+ace"],
+	"cbr": ["application/comicbook+rar"],
+	"cbt": ["application/comicbook+tar"],
+	"cbtc": ["application/comicbook+truecrypt"],
+	"cbz": ["application/comicbook+zip"],
```
Comment (rullzer) [inline resources/config/mimetypemapping.dist.json]: I was going over https://www.iana.org/assignments/media-types/media-types.xhtml

and there I can only find the mimetypes for the rar and zip one....

Comment (NoMoreNicksLeft) [reply resources/config/mimetypemapping.dist.json]: All the types in that file are in the original, my code only updates them to distinct mime values.

Wikipedia gives a few more hints, confirms that they're extant.

A post on reddit.com/r/datahoarder from someone who collects the files suggests that cbr is something like 95-97%, with the remainder being cbz, but that each is often misnamed the other.

I was not able to find any native cbtc to test against, I suspect it is mostly theoretical. 

The others follow the same pattern as rar and zip, and follow the same pattern you see on the IANA list.

Comment (rullzer) [reply resources/config/mimetypemapping.dist.json]: Ah fair enough.
I did notice that it seems to be `application/comicbook-rar` according to https://www.iana.org/assignments/media-types/media-types.xhtml

**Comment by MorrisJobke** [comment]
The mimetypes are available starting with Nextcloud 14 (see #10125). The previews then can be generated inside of an app. Thus I will close this PR here and ask you to create an app for this.

Thanks a lot for your contribution.

**Pull request closed without merging**